controllers/memcached/memcached_controller.go

```diff
@@ -256,3 +287,46 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Complete(r)
 }
+
+// GetServerLists returns list of memcached server without/with inet prefix
+func (r *Reconciler) GetServerLists(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *memcachedv1.Memcached,
+) ([]string, []string, error) {
+	var serverList []string
+	var serverListWithPrefix []string
+	var serverPrefix string
+
+	// Define the label selector to filter on
+	labelSelector := map[string]string{
+		"app":            "memcached",
+		"memcached/name": instance.Name,
+	}
+	podList, err := oko_pod.GetPodListWithLabel(ctx, h, instance.Namespace, labelSelector)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if len(podList.Items) == 0 {
+		return nil, nil, fmt.Errorf("No pods could be found")
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Spec.Hostname == "" || pod.Spec.Subdomain == "" || pod.Status.PodIP == "" {
+			return nil, nil, fmt.Errorf("Pod does not have the required Spec to construct endpoint")
+		}
```

Comment:
Similarly I would not treat this as an error in the reconciliation itself. This pod is probably not running yet, so you can simply ignore it here and process the rest of the pods.

Reply:
Fixed in the latest version